docs/metrics/wer-cer-mer.md

```diff
@@ -0,0 +1,196 @@
+    Candidate: `Amidst the emerald shadow, butterflies whistled secrets on the breeze.` <br>
+
+    <b>Word-level Substitutions</b>: <br>
+    <code>Amidst the emerald <b><big>shadow</big></b>, butterflies <b><big>whistled</big></b> secrets <b><big>on</big></b> the breeze.</code> <br>
+    <b>Character-level Substitutions</b>: <br>
+    <code>Amidst the emerald <b><big>sh</big></b>adow, butterflies <b><big>whistl</big></b>ed secrets <b><big>o</big></b>n the breeze.</code> <br>
+
+    In the above example, there are 3 word-level substitutions and 9 character-level substitutions.
+
+??? example "Deletion Errors"
+    Deletions occur when a candidate text is missing a word or sequence of words from the reference text.
+
+    <b> Example: </b> <br>
+    Reference: `Amidst the emerald meadow, butterflies whispered secrets in the breeze.` <br>
+    Candidate: `Amidst the emerald meadow, butterflies whispered.` <br>
+
+    <b>Word-level Deletions</b>: <br>
+    <code>Amidst the emerald meadow, butterflies whispered <u><big>secrets in the breeze</big></u>.</code> <br>
+    <b>Character-level Deletions</b>: <br>
+    <code>Amidst the emerald meadow, butterflies whispered <u><big>secrets in the breeze</big></u>.</code> <br>
+
+    In the above example, there are 4 word-level deletions and 18 character-level deletions.
+
+??? example "Insertion Errors"
+    Insertions occur when a candidate text contains an extra word or sequence of words that is not present in the reference text.
+
+    <b> Example: </b> <br>
+    Reference: `Amidst the emerald meadow, butterflies whispered secrets in the breeze.` <br>
+    Candidate: `Amidst the emerald meadow, butterflies whispered ethereal secrets in the breeze.` <br>
+
+    <b>Word-level Insertions</b>: <br>
+    <code>Amidst the emerald meadow, butterflies whispered <i><big>ethereal</big></i> secrets in the breeze.</code> <br>
+    <b>Character-level Insertions</b>: <br>
+    <code>Amidst the emerald meadow, butterflies whispered <i><big>ethereal</big></i> secrets in the breeze.</code> <br>
+
+    In the above example, there is 1 word-level insertion and 8 character-level insertions.
+
+Formally, it is defined as the rate of word-level errors in a candidate text.
+
+$$
+\text{WER} = \frac{\text{Substitutions} + \text{Deletions} + \text{Insertions}}{\text{# of Words in Reference}}
+$$
+
+### Example
+Let's calculate the word error rate between the following reference and candidate texts:
+
+| <b>Reference</b> | <b>Candidate</b> |
+| | |
+|  `The bard sang the ancient melodies of nature, transforming tranquil meadows into sonnets for enhanced soulful grace.` | `The poetic bard echoed the ancient melodies, transcending meadows into sonnets for soulful grace.` |
+
+??? example "Step 1. Count Errors"
+    Highlighting the <b><big>substitution</big></b>, <u><big>deletion</big></u>, and <i><big>insertion</big></i> errors, we can count each type of error:
```

Comment:
Instead of using bold, underline, and italic, should we colorcode these? I think that might be a better visualization.

Reply:
I've experimented with color coding and it becomes very hard to see in dark mode - should I keep it as formatted text or is there a workaround with dark mode?